microsoft/testsuites/virtualclient/virtualclient.py

```diff
@@ -0,0 +1,92 @@
+from typing import List, Optional, cast
+
+from lisa import (
+    Logger,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.features import Disk
+from lisa.operating_system import Debian
+from lisa.testsuite import TestResult
+from lisa.tools import VcRunner, VcTargetInfo
+
+
+@TestSuiteMetadata(
+    area="virtual_client",
+    category="performance",
+    description="""
+        This test suite runs the performance test cases with Virtual Client.
+    """,
+)
+class VirtualClient(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test is to run redis workload testing with Virtual Client.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            supported_os=[Debian],
+            min_count=2,
+        ),
+    )
+    def perf_vc_redis(self, log: Logger, result: TestResult) -> None:
+        self._run_work_load(result=result, profile_name="PERF-REDIS", roles=["client"])
+
+    @TestCaseMetadata(
+        description="""
+            This test is to run PostgreSQL workload testing with Virtual Client.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            supported_os=[Debian],
+            min_count=2,
+        ),
+        timeout=3000,
+    )
+    def perf_vc_postgresql(self, log: Logger, result: TestResult) -> None:
+        environment = result.environment
+        assert environment, "the environment should not be empty"
+        for node in environment.nodes.list():
+            disk = node.features[Disk]
+            disk.add_data_disk(count=1, size_in_gb=256)
+
+        self._run_work_load(
+            result=result,
+            profile_name="PERF-POSTGRESQL-HAMMERDB-TPCC",
+            timeout=45,
+        )
+
+    def _run_work_load(
```

Comment:
I think most of the method can be moved to vcrunner, like `def run(self, environment, profile_name, ...)`. The test methods need to specify roles and profile only.

Reply:
This method is used in this method only, the `system` is not set in test methods, so it should be removed.